clippy_lints/src/assign_ops.rs

```diff
@@ -237,6 +237,7 @@ fn is_commutative(op: hir::BinOpKind) -> bool {
     use rustc_hir::BinOpKind::{
         Add, And, BitAnd, BitOr, BitXor, Div, Eq, Ge, Gt, Le, Lt, Mul, Ne, Or, Rem, Shl, Shr, Sub,
     };
+    #[allow(clippy::match_like_matches_macro)]
```

Comment:
Would it make sense to only lint if the second arm matches on `PatKind::Wild`? That would make the lint a bit less noisy and prevent this FP, I think?

Reply:
I had initially considered that. However, that would mean a match with the arms `{Some(_), None}` would not be caught while `{Some(_), _}` would be despite them not being particularly different. I only hit one possibly intentionally exhaustive match in the entire clippy repository, and that `allow` serves as a (somewhat obscure) "hey, this is intentionally exhaustive".

Reply:
On second thought, `{Some(_), None}` will hit the `redundant_pattern_matching` lint anyway. It probably makes sense to restrict this lint to non-exhaustive matches.